src/brogue/MainMenu.c

```diff
@@ -260,136 +279,336 @@ void initializeMenuFlames(boolean includeTitle,
+    // The shading effect outside the rectangle decreases with distance.
+    // Warning: shading of neighboring rectangles stacks
+    rectangularShading(minX, minY, width, height + 1, &black, INTERFACE_OPACITY, shadowBuf);
+}
+
+/// @brief Initialize the main menu
+/// @param menu The main menu
+/// @param buttons The main menu buttons
+/// @param position The window position of the quit button
+/// @param shadowBuf The display buffer object for the background/shadow
+static void initializeMainMenu(buttonState *menu, brogueButton *buttons, windowpos position, cellDisplayBuffer shadowBuf[COLS][ROWS]) {
+    initializeMainMenuButtons(buttons);
+    stackButtons(buttons, MAIN_MENU_BUTTON_COUNT, position, 2, false);
+
+    initializeMenu(menu, buttons, MAIN_MENU_BUTTON_COUNT, shadowBuf);
+}
+
+/// @brief Initialize a flyout menu and position the buttons
+/// @param menu The menu to initialize
+/// @param shadowBuf The display buffer for the menu background/shadow
+/// @param buttons The buttons to add to the menu
+/// @param position The window position of the anchor button. All buttons are positioned relative to this location. 
+static void initializeFlyoutMenu(buttonState *menu, cellDisplayBuffer shadowBuf[COLS][ROWS], brogueButton *buttons, windowpos position) {
+    short buttonCount = 0;
+
+    if (rogue.nextGame == NG_FLYOUT_PLAY) {
+
+        buttonCount = 3;
+        initializeMainMenuButton(&(buttons[0]), "      %sN%sew Game     ", 'n','N', NG_NEW_GAME);
+        initializeMainMenuButton(&(buttons[1]), "  New %sS%seeded Game  ", 's','S', NG_NEW_GAME_WITH_SEED);
+        initializeMainMenuButton(&(buttons[2]), "     %sL%soad Game     ", 'l','L', NG_OPEN_GAME);
+
+    } else if (rogue.nextGame == NG_FLYOUT_VIEW) {
+
+        buttonCount = 2;
+        initializeMainMenuButton(&(buttons[0]), "   View %sR%secording  ", 'r','R', NG_VIEW_RECORDING);
+        initializeMainMenuButton(&(buttons[1]), "    %sH%sigh Scores    ", 'h','H', NG_HIGH_SCORES);
+
+    } else if (rogue.nextGame == NG_FLYOUT_OPTIONS) {
+
+        buttonCount = 2;
+        initializeMainMenuButton(&(buttons[0]), "    Game V%sa%sriant   ", 'a','A', NG_NOTHING);
+        initializeMainMenuButton(&(buttons[1]), "     Game %sM%sode     ", 'm','M', NG_GAME_MODE);
+
+    } else {
+        return;
+    }
+
+    stackButtons(buttons, buttonCount, position, 2, false);
+    initializeMenu(menu, buttons, buttonCount, shadowBuf);
+}
+
+/// @brief Displays a dialog window for the user to chose a game mode. The game mode is displayed in the bottom left
+/// on the title screen (except normal mode).
+static void chooseGameMode() {
+    short gameMode;
+    char textBuf[COLS * ROWS] = "", tmpBuf[COLS * ROWS] = "", goldColorEscape[5] = "", whiteColorEscape[5] = "";
+
+    encodeMessageColor(goldColorEscape, 0, &yellow);
+    encodeMessageColor(whiteColorEscape, 0, &white);
+
+    sprintf(tmpBuf,"%sNormal Mode%s\n",goldColorEscape,whiteColorEscape);
```

Comment:
I think spaces after commas is our usual style!

Reply:
Hard to catch these with just my eyeballs. I googled a regex that seems to work nicely though. `,(?=[^\s])`

I'll fix this and the others. Glad you like the text! I won't say how long I spent wordsmithing. 😄